static/js/notifications.js

```diff
@@ -402,6 +402,17 @@ function message_is_notifiable(message) {
         return false;
     }
 
+    // The notification should only be displayed once. To prevent multiple
+    // tabs from opening a notification each for their own, we store the
+    // notification ID in localStorage and check if a notification was
+    // already opened.
+    if (typeof(Storage) !== "undefined") {
```

Comment:
(Might be a good mini project to document how this all works as a new section in http://zulip.readthedocs.io/en/latest/front-end-build-process.html)